src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/BasicStatement.scala

```diff
@@ -84,7 +84,7 @@ private[querygeneration] object BasicStatement {
       case l: Literal =>
         l.dataType match {
           case StringType =>
-            if (l == null || l.toString() == "null") {
+            if (l.value == null) {
```

Comment:
Literal String `"null"` is regarded as NULL because `l.toString()` will be `"null"` for both below case:
1. `l` is a StringType and its value is NULL
2. `l` is a StringType and its value is string `"null"`